apps/src/applab/CrosshairOverlay.js

```diff
@@ -76,6 +77,13 @@ CrosshairOverlay.prototype.render = function (intoElement, nextProps) {
     rectY = this.props_.y - CROSSHAIR_MARGIN - TEXT_RECT_HEIGHT;
   }
 
+  // If we're dragging an element, instead put the text above and right of the
+  // cross hair, while making sure it doesnt go past the top of the overlay
+  if (this.props_.isDragging) {
```

Comment:
Any reason not to combine this case with the 'bottom of the visualization' case above?  The extra `Math.max` call isn't going to break the bank.  How about something like this?

```javascript
var rectY = this.props_.y + CROSSHAIR_MARGIN;
if (this.props_.isDragging || (rectY + TEXT_RECT_HEIGHT + EDGE_MARGIN > this.props_appHeight)) {
  // When dragging an element, or when the crosshair is near the bottom of the
  // visualization, put the coordinate text above the crosshair instead.
  rectY = Math.max(0, this.props_.y - CROSSHAIR_MARGIN - TEXT_RECT_HEIGHT);
}
```